### DISCUSSION TITLE
Interpreting info displayed in safety mode

### DISCUSSION THREAD

**Discussion by teeten** [jmcollin78/versatile_thermostat#335] [Q&A] [answered]
<img width="437" alt="image" src="https://github.com/jmcollin78/versatile_thermostat/assets/24637509/ef058f53-f4b8-4931-9919-4b37a86d3f65">

Why does my thermostat remain in safety mode even though the room temperature sensor reported a change less than 1 hour ago?

**Comment by jmcollin78** [comment]
The answer is in the screen copy 😄 . The room temperature sensor is present. Yes it have report temperature for 26 min.
But what about the external temperature sensor ?

**Reply by jmcollin78** [reply]
You can also consult this and the README : https://github.com/jmcollin78/versatile_thermostat/discussions/261

**Reply by teeten** [reply]
Thanks for your answer. I have read the documentation several times (in French and English :smile:).

I am using a weather temperature sensor, so I have no control over its update frequency, and because of that, my thermostat goes into safety mode unnecessarily (because, with the current freezing temperatures and a badly insulated room, the duty cycle goes up to 60-70% easily).

Can you help me understand why the external sensor is taken into account for safety mode? If the room sensor is working but not the external one, I assume that the heating will still stop when the target room temperature is reached, so what is the risk?

Any idea of how I could work around these issues?

**Reply by jmcollin78** [reply]
Ne serais-tu un certain [Swiiney](https://forum.hacf.fr/u/Swiiney) qui m'a posé exactement la même question dans le Forum hacf ?
(et auquel j'ai déjà répondu).
Si oui l'intégration Meteo France fonctionne très bien pour avoir la température extérieure. J'ai jamais (rarement ?) eu de soucis avec.

**Comment by teeten** [comment]
Non je ne suis pas Swiiney! Je suis un utilisateur relativement récent de Home Assistant mais j'ai décidé de m'y mettre sérieusement. Etant ingénieur software de métier avec quelques heures de vols à  mon actif 👴, je finirai peut-être par coder 😉 ... mais pour le moment j'étudie et j'apprends (avec ton tutorial)! J'ai testé plusieurs thermostats (dont le générique et le smart) et le tien est celui qui se rapproche le plus des spécifications que j'aurais écrites intuitivement (avec mes modestes bases en thermo 🙇‍♂️). Ma priorité pour démarrer en domotique était le contrôle du chauffage, et grâce à ton thermostat, ça marche plutôt pas mal (encore merci).

Pour revenir à ma question initiale, je n'ai aucun souci non plus avec l'intégration Météo France, et je ne ferais pas mieux avec mon propre thermomètre. Mais comme tu le vois sur ma copie d'écran, il est arrivé que la température remontée par l'intégration ne change pas pendant plus de 2h, justement au moment où mes radiateurs étaient à plus de 60%. J'admets que la probabilité est faible et que ce n'est pas un drame, mais je me dis que tu as probablement une bonne raison de prendre en compte la température extérieure pour le mode sécurité, du coup je suis curieux de connaître le "rationale", comme on dit dans notre jargon.

Je pars donc à la recherche de la réponse que tu as faite à Swiiney ... 🤞  - JC (Suresnes, 92)

**Reply by jmcollin78** [reply, marked as answer]
ok pas de soucis. La simultanéité et la ressemblance de la question m'a troublé.

Merci pour les compliments !

La température extérieure sert au calcul (cf. l'algo TPI) de la puissance qu'on va demander au radiateur. Donc si elle est fausse, on prend le risque de demander au radiateur de chauffer alors qu'il ne le faudrait pas (ou l'inverse) et comme la boucle de retroaction est cassée, on peut avoir un système divergent. Au pire, ça se finit potentiellement en incendie si il n'y a aucune sécurité sur le radiateur lui-même.

Mais ...

Comme c'est un soucis cette histoire de température extérieure qui ne bouge plus et que ça coupe potentiellement tous les radiateurs si ca arrive, je suis en train de mettre en place un débrayage de ce check. C'est là : https://github.com/jmcollin78/versatile_thermostat/issues/343

C'est développé, en test chez moi et devrait être publié ce week-end. Tu pourras (sous ta responsabilité) prendre le risque de désactiver cette fonction. Si le coef_ext est faible, ca ne pose pas plus de soucis que ça. Si le coef_ext est fort, c'est dangereux. Donc ce sera une fonction un peu cachée (en mode "expert" pour reprendre le vocabulaire) et pas à la main de tout le monde.

T'as 3 moyens de mitiger le mode sécurité:
- le délai d'indispo des thermomètres,
- le taux de chauffe minimum avant qu'on active le mode sécurité (ca sert à rien de mettre en sécurité si on chaufffe a 10%),
- le taux de chauffe "remplaçant" (si le mode sécurité est activé, on ne coupe pas totalement la chauffe mais on met un peu de pourcentage, jugé non dangereux, qui permet de ne pas avoir une maison à 5° lorsque ca se produit).

Avec ces 3 paramètres, on arrive quasiment à réduire à zéro le nombre de déclenchements intempestifs. Si tu mets 2 h, 60% et 30%. Ca ne se déclenchera qu'au bout de 2 h d'indispo (c'est rare quand même), si le taux de chauffe est supérieur à 60% et ça mettra un taux de chauffe à 30% en remplacement du 60%.

**Comment by teeten** [comment]
Merci pour ta réponse (quelle rapidité 😄 !), parfaitement claire. J'ai bien vu et compris la formule du TPI.

J'avais effectivement pris ces 3 mesures de mitigation: délai d'indisponibilité de 2h, pourcentage à 80% (mes radiateurs sont un peu sous-dimensionnés par rapport à mon isolation, travaux en cours ...), et rapport cyclique à 20% en mode sécurité.

Je suis d'accord qu'avec un coefficient extérieur faible, le risque est limité (j'ai tout de même dû passer à 0.03 pour une chambre très mal isolée, en raison du temps de ces derniers jours). D'ailleurs, avec ce coefficient à zéro, je suppose que le thermostat fonctionne à peu près bien mais en restant sous la consigne, n'est-ce pas?

Je jetterai un coup d'oeil au nouveau mode expert après la mise à jour, merci pour l'annonce anticipée 😉 .

**Comment by teeten** [comment]
Pour info, j'ai fait un commentaire sur https://github.com/jmcollin78/versatile_thermostat/issues/343.
